HeinOnline.js

```diff
@@ -0,0 +1,210 @@
+
+    Zotero is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+    ***** END LICENSE BLOCK *****
+*/
+
+var listItems;
+
+function detectWeb (doc, url) {
+	if (url.indexOf("/LuceneSearch?") > -1) {
+		if (getSearchResults(doc)) {
+			return "multiple";
+		}
+	} else {
+		return "journalArticle";
+	}
+	return false;
+}
+
+function getXPathStr(attr, elem, path) {
+	var res = ZU.xpath(elem, path);
+	res = res.length ? res[0][attr] : '';
+	return res ? res : '';
+}
+
+function Data(doc) {
+	this.node = ZU.xpath(doc, '//form[@id="Print1"]');
+	this.urlbase = "PDFsearchable?sectioncount=1&ext=.pdf&nocover=&";
+	this.queryElems = [];
+}
+
+Data.prototype.getVal = function(name, returnOnly) {
+
+	var val = getXPathStr("value", this.node, './/input[@name="' + name + '"]');
+	val = val ? encodeURIComponent(val) : '';
+
+	if (!returnOnly) {
+		this.queryElems.push(name + "=" + val);
+	}
+	return val;
+}
+
+Data.prototype.dump = function() {
+	return this.urlbase + this.queryElems.join("&");
+}
+
+function getSearchResults(doc) {
+	var results = doc.getElementsByClassName("lucene_search_result_b"),
+		items = {},
+		found = false
+	for (var i=0; i<results.length; i++) {
+		var url = getXPathStr("href", results[i], './/a[contains(@href, "Print")]');
+		url = url.replace(/Print/, "Page");
+		url = url.replace(/&terms=[^&]*/, '');
+
+		var title = getXPathStr("textContent", results[i], './/a[1]');
+		title = ZU.trimInternal(title);
+		title = title.replace(/\s*\[[^\]]*\]$/, '');
+
+		if (!title || !url) continue;
+		
+		items[url] = title;
+		found = true;
+	}
+	return found ? items : false;
+}
+
+function scrapePage(doc, url) {
+	var pdfPageURL = url.replace(/\/Page\?/, "/Print?");
+	var item = new Zotero.Item();
+	var z3988title = getXPathStr("title", doc, '//span[contains(@class, " Z3988") or contains(@class, "Z3988 ") or @class="Z3988"][@title]');
+	ZU.parseContextObject(z3988title, item);
+	
+	if (!item.itemType) {
+		// Sometimes items that report full-text PDF in the search listing
+		// resolve to a failure page. This builds a placeholder item out
+		// of the data that is available.
+		item.itemType = "journalArticle";
+		if (listItems) {
+			item.title = listItems[url];		
+		} else {
+			item.title = getXPathStr("textContent", doc, '//div[@id="content-container"]//a[1]');
+		}
+		var notAvailableURL = url.replace("/Page?", "/NotAvailable?").replace("handle=", "handle_bad=");
+		item.attachments.push({
+			url:notAvailableURL,
+			mimetype:"text/html",
+			snapshot:true,
+			title:"HeinOnline page placeholder"
+		});
+		item.url = url;
+		item.abstract = "CAUTION: Resource not yet available at HeinOnline";
```

Comment:
I agree that `abstractNote` is the wrong field to save any dummy text which can be used as a flag. Maybe, a note might be better?